Chapter2/README.md

```diff
@@ -0,0 +1,305 @@
+
+#### Configure the  service app (TodoListAPI) to use your app registration
+
+Open the project in your IDE (like Visual Studio) to configure the code.
+>In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `TodoListAPI\appsettings.json` file.
+1. Find the app key `Domain` and replace the existing value with your Azure AD tenant name.
+1. Find the app key `ClientId` and replace the existing value with the application ID (clientId) of the `TodoListAPI` application copied from the Azure portal.
+
+### Register the client app (TodoListSPA)
+
+1. Navigate to the Microsoft identity platform for developers [App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page.
+1. Select **New registration**.
+1. In the **Register an application page** that appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `TodoListSPA`.
+   - Under **Supported account types**, select **Accounts in any organizational directory and personal Microsoft accounts**.
+   - In the **Redirect URI** section, select **Single-page application** in the combo-box and enter the following redirect URI: `http://localhost:4200/`.
+1. Select **Register** to create the application.
+1. In the app's registration screen, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select **Authentication** in the menu.
+   - If you don't have a platform added, select **Add a platform** and select the **Single-page application** option.
+   - In the **Implicit grant** section, check the **Access tokens** and **ID tokens** option as this sample requires
+     the [Implicit grant flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to be enabled to
+     sign-in the user, and call an API.
+
+1. Select **Save** to save your changes.
+1. In the app's registration screen, click on the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+   - Click the **Add a permission** button and then,
+     - Ensure that the **Microsoft APIs** tab is selected
+     - In the list of APIs, select the API **Microsoft Graph**.
+     - In the **Delegated permissions** section, select the **User.Read.All** in the list. Use the search box if necessary.
+     - Click on the **Add permissions** button at the bottom.
+   - Click the **Add a permission** button again and then,
+     - Ensure that the **My APIs** tab is selected.
+     - In the list of APIs, select the API `TodoListAPI`.
+     - In the **Delegated permissions** section, select the **access_as_user** in the list. Use the search box if necessary.
+     - Click on the **Add permissions** button at the bottom.
+
```

Comment:
Should we update the heading here back to TodoListAPI for clarity?

Reply:
Makes sense actually. Ok I'll revise.

Reply:
Done! Added a warning box to draw attention.